backend/src/flotilla/database/crud.py

```diff
@@ -31,6 +34,64 @@ def read_robot_by_id(db: Session, robot_id: int) -> RobotDBModel:
     return robot
 
 
+def create_robot(
+    db: Session,
+    name: str,
+    model: str,
+    serial_number: str,
+    host: str,
+    port: int,
+    enabled: bool,
+    capabilities: List[str],
+) -> int:
+
+    robot: RobotDBModel = RobotDBModel(
+        name=name,
+        model=model,
+        serial_number=serial_number,
+        host=host,
+        port=port,
+        status=RobotStatus.available,
+        enabled=enabled,
+    )
+
+    _check_non_duplicate_robot(db=db, robot=robot)
+
+    _add_capabilities_to_robot(robot=robot, capabilities=capabilities)
+
+    db.add(robot)
+    db.commit()
+
+    return robot.id
+
+
+def _add_capabilities_to_robot(robot: RobotDBModel, capabilities: List[str]) -> None:
+    for capability in capabilities:
+        if not InspectionType.has_value(capability):
+            raise HTTPException(
+                status_code=HTTPStatus.BAD_REQUEST.value,
+                detail="Invalid format for robot capabilities",
+            )
+
+        CapabilityDBModel(robot=robot, capability=InspectionType(capability))
```

Comment:
Does this work? Thought the CapabililityDBModel wanted robot_id and not the whole robot. Some sort of sqlalchamy magic?

Reply:
Yes, it does work. However, I found an issue when attempting to do the same in some locations in the mock database setup. I'll experiment a bit more first.